crates/log/src/setup.rs

```diff
@@ -0,0 +1,44 @@
+use bevy::prelude::*;
+use tracing::Level;
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::{fmt, EnvFilter};
+
+pub(crate) struct LogPlugin {}
+
+/// the handle for the guard (dropping it will disable the log writer)
+#[derive(Resource)]
+pub(crate) struct CurrentLogHandle {
+    #[allow(dead_code)]
+    guard: WorkerGuard,
+}
+
+impl Plugin for LogPlugin {
+    fn build(&self, app: &mut App) {
+        // for file name
+        let dt = chrono::Local::now();
+
+        let file_name = dt.format("%Y-%m-%d %H:%M:%S.log").to_string();
```

Comment:
* I would change the file name so it does not contain a space (to make it easier to work with them in the command line) or a colon (this causes issues on MacOS as I was told).

* Opinionated nitpick: I would make `file_name` a `PathBuf` so that the trick `&*file_name` is not needed: `let path: PathBuf = dt.format("%Y-%m-%dT%H%M%S.log").to_string().into();`

Reply:
Would this format be good? %Y-%m-%d_%H-%M-%S.log (2023-05-09_10-13-17.log)